custom/icds_reports/management/commands/dashboard_usage_report.py

```diff
@@ -0,0 +1,178 @@
+
+        # fetching username to state mapping
+        username_vs_state_name = defaultdict()
+        for data in usage_data[1:]:
+            username = data[4]
+            state_name = data[1]
+            if username not in username_vs_state_name.keys():
+                username_vs_state_name[username] = state_name
+
+        self.get_tabular_data(usage_data, start_date, end_date, domain)
+        print(f'Request data written to file {TABULAR_DATA_CACHE}')
+        self.get_cas_data(username_vs_state_name, start_date, end_date, domain)
+        print(f'Request data written to file {CAS_DATA_CACHE}')
+
+    def get_dashboard_tabular_usage_counts(self, start_date, end_date, domain):
+        """
+        :param start_date: start date of the filter
+        :param end_date: end date of the filter
+        :param domain
+        :return: returns the counts of no of downloads of each and total reports for  all usernames
+        """
+        print(f'Compiling tabular usage counts for users')
+        tabular_user_counts = defaultdict(int)
+        tabular_user_indicators = defaultdict(lambda: [0] * 10)
+
+        records = list(ICDSAuditEntryRecord.objects.filter(url=f'/a/{domain}/icds_export_indicator',
+                                                           time_of_use__gte=start_date,
+                                                           time_of_use__lte=end_date)
+                       .annotate(indicator=Cast(KeyTextTransform('indicator', 'post_data'), IntegerField()))
+                       .filter(indicator__lte=THR_REPORT_EXPORT).values('indicator', 'username')
+                       .annotate(count=Count('indicator')).order_by('username', 'indicator'))
+        for record in records:
+            tabular_user_counts[record['username'].split('@')[0]] += record['count']
+            tabular_user_indicators[record['username'].split('@')[0]][int(record['indicator']) - 1]\
+                = record['count']
+
+        return tabular_user_counts, tabular_user_indicators
+
+    def get_dashboard_cas_usage_counts(self, start_date, end_date, domain):
+        """
+        :param start_date: start date of the filter
+        :param end_date: end date of the filter
+        :param domain
+        :return: returns the counts of no of downloads of each and total reports for  all usernames
+        """
+        print(f'Compiling cas export usage counts for users')
+        cas_user_counts = defaultdict(int)
+
+        records = list(ICDSAuditEntryRecord.objects.filter(url=f'/a/{domain}/cas_export',
+                                                           time_of_use__gte=start_date,
+                                                           time_of_use__lte=end_date)
+                       .annotate(indicator=Cast(KeyTextTransform('indicator', 'post_data'), IntegerField()))
+                       .filter(indicator__lte=THR_REPORT_EXPORT).values('indicator')
```

Comment:
You dont need this indicator_lte filter for cas usage count.

Reply:
removed